lib/jit_preloader/active_record/associations/preloader/ar5_association.rb

```diff
@@ -18,6 +18,8 @@ module PreloaderAssociation
     # end
 
     def run(preloader)
+      return unless (reflection.scope.nil? || reflection.scope.arity == 0) && klass.ancestors.include?(ActiveRecord::Base)
```

Comment:
Can you explain what `reflection.scope.arity == 0` is checking for in this context? I'm a bit confused why the arity of the scope would impact `run`.

Reply:
My understanding is that scopes whose arity is non-zero are scopes which require an argument. Those scopes are not automatically preloadable because we can't guess at what the argument is supposed to be, so we skip trying to preload them.

Reply:
Thanks for the explanation!

Reply:
For reference, here's the equivalent code from Rails master: https://github.com/rails/rails/blob/main/activerecord/lib/active_record/reflection.rb#L478-L489